react/components/EXPERIMENTAL_Table/index.tsx

```diff
@@ -18,20 +19,29 @@ const propTypes = {
   items: PropTypes.arrayOf(PropTypes.object),
   isEmpty: PropTypes.bool,
   tableHeight: PropTypes.number,
+  rowHeight: PropTypes.number,
+  selectedDensity: PropTypes.oneOf(['low', 'medium', 'high']),
```

Comment:
Could those value be called from an external const? I've seen those written already somewhere else.

Reply:
The states are controlled by the `useTableState` hook that is used outside of the `Table` itself. So, the parent has full control of these values before passing them as props. It is very useful if the user wants to take actions based on these values. Do you agree with the usage or that is best to call the hook inside of `Table` component  (this will reduce the props to `columns`, `items` and `density`)?

Reply:
I find very useful the values outside, because instead of overloading the table with callback props such as `onToggleDensity`, `onHiddeFields`, `setSetDensityToLow`, `onWhatever` I can just use an `effect` for it. Such as: 

```js
const tableState = useTableState(...);

useEffect( ( ) => {
  if (tableState.selectedDensity === 'low') {  /* do my magic */ }
}, [tableState.selectedDensity])

<Table {...tableState} />
```
It's verry powerful and avoids 50 milion abstractions.
And I do think that a prop called `state` that receives all the states at the table, would look better (this spreading is not that appealing and can cause confusion). Such as:  `<Table state={tableState} />`

Would like to hear your thoughts on this. 🙂